example/src/new_api/swipeable/Swipeable.tsx

```diff
@@ -0,0 +1,573 @@
+   * progressAnimatedValue: [0, 1] dragAnimatedValue: [0, +]
+   *
+   * To support `rtl` flexbox layouts use `flexDirection` styling.
+   * */
+  renderLeftActions?: (
+    progressAnimatedValue: SharedValue<number>,
+    dragAnimatedValue: SharedValue<number>,
+    swipeable: ForwardedRef<ExposedFunctions> // we have to use ref here, as it now holds all the objects
+  ) => React.ReactNode;
+  /**
+   *
+   * This map describes the values to use as inputRange for extra interpolation:
+   * AnimatedValue: [startValue, endValue]
+   *
+   * progressAnimatedValue: [0, 1] dragAnimatedValue: [0, -]
+   *
+   * To support `rtl` flexbox layouts use `flexDirection` styling.
+   * */
+  renderRightActions?: (
+    progressAnimatedValue: SharedValue<number>,
+    dragAnimatedValue: SharedValue<number>,
+    swipeable: ForwardedRef<ExposedFunctions>
+  ) => React.ReactNode;
+
+  useNativeAnimations?: boolean;
+
+  animationOptions?: Record<string, unknown>;
+
+  /**
+   * Style object for the container (`Animated.View`), for example to override
+   * `overflow: 'hidden'`.
+   */
+  containerStyle?: StyleProp<ViewStyle>;
+
+  /**
+   * Style object for the children container (`Animated.View`), for example to
+   * apply `flex: 1`
+   */
+  childrenContainerStyle?: StyleProp<ViewStyle>;
+}
+
+export interface ExposedFunctions {
+  close: () => void;
+  openLeft: () => void;
+  openRight: () => void;
+  reset: () => void;
+}
+
+const Swipeable = forwardRef<ExposedFunctions, SwipeableProps>(
+  function Swipeable(
+    props: SwipeableProps,
+    ref: ForwardedRef<ExposedFunctions>
+  ) {
+    const rowState = useSharedValue<number>(0);
+
+    const dragX = useSharedValue<number>(0);
+    const rowTranslation = useSharedValue<number>(0);
+    const leftWidth = useSharedValue<number>(0);
+    const rightOffset = useSharedValue<number>(0);
+    const rowWidth = useSharedValue<number>(0);
+
+    // todo: if things don't work, set all the default values from the original file
+    // they are set everywhere thoughout this file, many functions set the default states themselves
+
+    const defaultProps = {
+      friction: 1,
+      overshootFriction: 1,
+      useNativeAnimations: true,
+    };
+
+    useEffect(() => {
+      if (!props.friction) props.friction = defaultProps.friction;
+      if (!props.overshootFriction)
+        props.overshootFriction = defaultProps.overshootFriction;
+      if (!props.useNativeAnimations)
+        props.useNativeAnimations = defaultProps.useNativeAnimations;
+
+      updateAnimatedEvent();
+    }, []);
+
+    /* todo: export to the exposed store
+  
+  const shouldComponentUpdate = (props: SwipeableProps) => {
+    if (
+      props.friction !== props.friction ||
+      props.overshootLeft !== props.overshootLeft ||
+      props.overshootRight !== props.overshootRight ||
+      props.overshootFriction !== props.overshootFriction
+    ) {
+      updateAnimatedEvent(props);
+      return true;
+    }
+    return false;
+  };
```

Comment:
I think it would be good to deprecate this function in favour of `onUpdate` callback which would be provided by the user. This would be more aligned with how other react components work. Would that be alright? @j-piasecki

Reply:
this was an internal function all along, my mistake